File: frontend/lib/models/payment.dart
class Payment {
  final String id;
  final String houseId;
  final String fromUid;
  final String fromName;
  final String toUid;
  final String toName;
  final double amount;
  final DateTime date;

  Payment({
    required this.id,
    required this.houseId,
    required this.fromUid,
    required this.fromName,
    required this.toUid,
    required this.toName,
    required this.amount,
    required this.date,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['id'],
      houseId: json['houseId'],
      fromUid: json['fromUid'],
      fromName: json['fromName'],
      toUid: json['toUid'],
      toName: json['toName'],
      amount: double.parse(json['amount'].toString()),
      date: DateTime.parse(json['date']),
    );
  }
}

File: backend/src/main/java/com/casacontrole/model/Payment.java
package com.casacontrole.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Payment {
    private String id;
    private String houseId;
    private String fromUid;
    private String fromName;
    private String toUid;
    private String toName;
    private BigDecimal amount;
    private Instant date;
    private Instant createdAt;
}
